logic/path/src/selectors/legacy.rs

```diff
@@ -22,23 +27,51 @@ impl PartialOrd for WeightedChannelPath {
 }
 
 impl Ord for WeightedChannelPath {
+    /// Favors not expored paths over fully explored paths,
+    /// there could be better ones.
+    ///
+    /// Favors longer paths over shorter paths, longer path
+    /// means more privacy.
+    ///
+    /// If both parts are of the same length, favors paths
+    /// with higher weights.
     fn cmp(&self, other: &Self) -> Ordering {
-        self.1.cmp(&other.1)
+        if other.fully_explored == self.fully_explored {
+            match self.path.len().cmp(&other.path.len()) {
+                Ordering::Equal => self.weight.cmp(&other.weight),
+                Ordering::Greater => Ordering::Greater,
+                Ordering::Less => Ordering::Less,
+            }
+        } else if other.fully_explored && !self.fully_explored {
+            Ordering::Less
+        } else {
+            Ordering::Greater
+        }
     }
 }
 
-/// Calculates the channel weight as (channel stake + 1) * (1 + R)
-/// where R is uniform random in (0,1]
+/// Assigns each channel a weight.
+/// The weight is randomized such that not always the same
+/// nodes get selected. This is necessary to achieve privacy.
+/// It also favors nodes with higher stake.
 pub struct RandomizedEdgeWeighting;
 
 impl EdgeWeighting<U256> for RandomizedEdgeWeighting {
+    /// Multiply all channel stake with a random float in the interval (0,1].
+    /// Given that the floats are uniform, nodes with higher stake have a higher
+    /// probability of reaching a higher value.
+    ///
+    /// Sorting the list of weights thus moves nodes with higher stakes more
+    /// often to the front.
     fn calculate_weight(channel: &ChannelEntry) -> U256 {
-        const PATH_RANDOMNESS: f64 = 0.1;
-
-        let r = random_float() * PATH_RANDOMNESS;
-        let base = channel.balance.amount() + 1;
-
-        base.add(base.mul_f64(r).unwrap())
+        max(
+            U256::one(),
+            channel
+                .balance
+                .amount()
+                .mul_f64(random_float())
+                .expect("Could not multiply edge weight with float"),
```

Comment:
I'm not confident this is the best solution, why abort the entire node, because the multiplication is not possible, which can happen:
```suggestion
                .unwrap_or(U256::max()),
```

Reply:
handled